research-hub-web/src/app/components/shared/body-media/expandable-page-part/expandable-page-part.component.ts

```diff
@@ -0,0 +1,94 @@
+import { Component, Input, OnDestroy, OnInit, Type } from '@angular/core';
+import { Expand, GetExpandPartByIdGQL, Scalars } from '@app/graphql/schema';
+import { BodyMediaService } from '@services/body-media.service';
+import { NodeRenderer, MarkRenderer } from 'ngx-contentful-rich-text';
+import { Observable, Subscription } from 'rxjs';
+import { pluck } from 'rxjs/operators';
+
+/**
+ * Exandable page part is a special type of rich text entry which itself contains rich text. For this reason it is
+ * handled differently from the other rich text types. It is a type of embedded entry block (see BlocksEmbeddedEntryComponent), so
+ * the top level node is handled in that component, but the nested rich text contained within the expandable part is handled here.
+ * The BlocksEmbeddedEntryComponent passes the ID for the expand part to this component, where we can then fetch the rich text from the 
+ * graphql API. This also avoids any nested/recursive calls (e.g. there could be an expandable part within an expandable part) to the api.
+ */
+@Component({
+  selector: 'app-expandable-page-part',
+  templateUrl: './expandable-page-part.component.html',
+  styleUrls: ['./expandable-page-part.component.scss']
+})
+export class ExpandablePagePartComponent implements OnInit, OnDestroy {
+  @Input() contentItem: Partial<Expand>; // basically just the id of the expand e.g. { "__typename": "Expand", "sys": { "id": "3sOzm7PRhsgcGibaW73EXN", "__typename": "Sys" } }
+  
+  public expandPart$: Observable<Expand>;
+  private subscriptions = new Subscription();
+  
+  public nodeRenderers: Record<string, Type<NodeRenderer>>;
+  public markRenderers: Record<string, Type<MarkRenderer>>;
+
+  public panelOpenState = false;
+
+  constructor(
+    private getExpandPartByIdGQL: GetExpandPartByIdGQL,
+    public bodyMediaService: BodyMediaService,
+  ) {
+    this.nodeRenderers = this.bodyMediaService.nodeRenderers;
+    this.markRenderers = this.bodyMediaService.markRenderers;
+  }
+
+  ngOnInit(): void {
+    this.expandPart$ = this.getExpandPart();
+    this.subscriptions.add(this.expandPart$.subscribe(data => {
+      // For each rich text field add the links to the link maps in the body media service to enable rich text rendering
+      this.bodyMediaService.buildLinkMaps(data.bodyText?.links);
+    }));
+  }
+
+  private getExpandPart(): Observable<Expand> {
+    try {
+      return this.getExpandPartByIdGQL.fetch({ id: this.contentItem.sys.id }).pipe(
+        pluck('data', 'expand')
+      ) as Observable<Expand>;
+    } catch (e) {
+      console.error(`Error loading expandable page part ${this.contentItem.sys.id}:`, e);
+    }
+  }
```

Comment:
I think the `try/catch` in `getExpandPart()` doesn't actually do anything in case there is no result, for example if the ID is unknown (it's probably an unlikely scenario, but better safe than sorry). In the current implementation `getExpandPart()` would return `null` and the subscriber would just pass that into `buildLinkMaps`. I think checking if the expand part exists, throwing an Error if it doesn't and then handling the error in the subscription is a safer way. I also allows us to handle the error in whatever way we want int the subscriber.

This is how I'm handling the requests in the content page refactor as well.

```suggestion
  ngOnInit(): void {
    this.expandPart$ = this.getExpandPart();
    this.subscriptions.add(this.expandPart$.subscribe({
      next: data => {
        // For each rich text field add the links to the link maps in the body media service to enable rich text rendering
        this.bodyMediaService.buildLinkMaps(data.bodyText?.links);
      },
      error: (error) => console.error(error) // handle error here
    }));
  }

  private getExpandPart(): Observable<Expand> {
    return this.getExpandPartByIdGQL.fetch({ id: this.contentItem.sys.id }).pipe(
      map((result) => {
        if (result.data.expand) {
          return result.data.expand as Expand
        } else {
          throw new Error(`Could not find expandable page part for id ${this.contentItem.sys.id}`)
        }
      })
    );
  }
```